File: frontend/app/src/main/java/com/example/commov/viewmodel/DashboardUiState.java
package com.example.commov.viewmodel;

import com.example.commov.model.DashboardTask;

import java.util.List;

public final class DashboardUiState {
    public final String userName;
    public final int pendingTasks;
    public final int completedTasks;
    public final int pendingProgress;
    public final int completedProgress;
    public final List<DashboardTask> tasks;

    public DashboardUiState(
            String userName,
            int pendingTasks,
            int completedTasks,
            int pendingProgress,
            int completedProgress,
            List<DashboardTask> tasks
    ) {
        this.userName = userName;
        this.pendingTasks = pendingTasks;
        this.completedTasks = completedTasks;
        this.pendingProgress = pendingProgress;
        this.completedProgress = completedProgress;
        this.tasks = tasks;
    }
}

File: frontend/app/src/main/java/com/example/commov/viewmodel/DashboardUiState.kt
package com.example.commov.viewmodel

import com.example.commov.model.DashboardTask
import com.example.commov.model.Project

data class DashboardUiState(
    val userName: String,
    val todoTasks: Int,
    val completedTasks: Int,
    val todoProgress: Int,
    val completedProgress: Int,
    val tasks: List<DashboardTask>,
    val overdueTasks: List<DashboardTask>,
    val todayTasks: List<DashboardTask>,
    val weekTasks: List<DashboardTask>,
    val projects: List<Project>,
    val hoursLoggedThisWeek: Double,
    val tasksOverEstimate: Int,
    val canManageProjects: Boolean,
    val requiresLogin: Boolean,
    val isLoading: Boolean = true,
    val isRefreshing: Boolean = false
)
